src/main/java/org/ow2/proactive/procci/request/CloudAutomationInstances.java

```diff
@@ -0,0 +1,121 @@
+package org.ow2.proactive.procci.request;
+
+import java.util.Optional;
+
+import org.ow2.proactive.procci.model.ModelConstant;
+import org.ow2.proactive.procci.model.cloud.automation.Model;
+import org.ow2.proactive.procci.model.exception.CloudAutomationException;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+
+/**
+ * Created by mael on 02/06/16.
+ */
+
+/**
+ * Manage the connection and the request with Cloud Automation Microservices
+ */
+
+public class CloudAutomationInstances {
+
+    private final Logger logger = LogManager.getRootLogger();
+
+    /**
+     * Get the deployed instances from Cloud Automation Model
+     *
+     * @return a json object containing the request results
+     */
+    public JSONObject getRequest() throws CloudAutomationException {
+        final String url = RequestUtils.getInstance().getProperty(
+                "cloud-automation-service.instances.endpoint");
+        JSONObject result = new JSONObject();
+        try {
+            CloseableHttpClient httpClient = HttpClientBuilder.create().build();
+            HttpGet getRequest = new HttpGet(url);
+
+            HttpResponse response = httpClient.execute(getRequest);
+            String serverOutput = RequestUtils.readHttpResponse(response);
+            httpClient.close();
+            result = (JSONObject) new JSONParser().parse(serverOutput);
+        } catch (Exception ex) {
+            raiseException(ex);
+        }
+
+        return result;
+    }
+
+
+    /**
+     * Get the instance of cloud automation service and return the first occurance with variableValue matching variableName
+     *
+     * @param variableName  a key in variables
+     * @param variableValue the value matching with the variableName key
+     * @return the first occurance which match with variablename and variableValue
+     */
+    public Optional<Model> getInstanceByVariable(String variableName,
+            String variableValue) throws CloudAutomationException {
+        JSONObject instances = getRequest();
+
+        return instances.keySet()
+                .stream()
+                .map(key -> ((JSONObject) instances.get(key)).get(ModelConstant.VARIABLES))
+                .filter(vars -> ((JSONObject) vars).containsKey(variableName))
```

Comment:
it will not improve the performance because the transformation are needed for each filtering.